### PR TITLE
add page for the official canto dex

### PR DESCRIPTION
**Pull request by tushar994**

### CODE REVIEW

File: src/static/js/canto.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const main = async() => {
     "title":"CANTO Network",
     "heading":["Pool Provider","MCN","LP", "Reward Tokens", "INFO"],
     "rows": [
-      ["Encanto       ","", `<a href='encanto'       >Various</a>`,"ENCANTO          ","https://encantowetrust.com"]
+      ["Encanto       ","", `<a href='encanto'       >Various</a>`,"ENCANTO          ","https://encantowetrust.com"],
+      ["Canto         ","", `<a href='canto'         >Various</a>`,"CANTO            ","                          "]
```
Comment (zjesko) [inline src/static/js/canto.js]: Rename to `Canto Dex`

File: src/static/js/canto_canto.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
+
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: rename file to `canto_dex.js`

File: src/static/js/canto_canto.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
...:"","type":"bool"}],"name":"getPair","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"","type":"address"}],"name":"isPair","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"isPaused","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"pairCodeHash","outputs":[{"internalType":"bytes32","name":"","type":"bytes32"}],"stateMutability":"pure","type":"function"},{"inputs":[],"name":"pauser","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"pendingPauser","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"bool","name":"_state","type":"bool"}],"name":"setPause","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"_pauser","type":"address"}],"name":"setPauser","outputs":[],"stateMutability":"nonpayable","type":"function"}]
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: rename `CANTO_FACTORY` to `BaseV1Factory` and similar

File: src/static/js/canto_canto.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    ]
+
+    async function main() {
+        const App = await init_ethers();
+    
+        _print(`Initialized ${App.YOUR_ADDRESS}\n`);
+        _print("Reading smart contracts...\n");
+    
+       const CANTO_FACTORY_ADDR = "0xE387067f12561e579C5f7d4294f51867E0c1cFba";
+       const rewardTokenTicker = "  ";
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: put `wCANTO` here

File: src/static/js/canto_canto.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        for(let k in prices) {
+            prices[k.toLowerCase()] = prices[k];
+        }
+
+        const poolCount = pairAddresses.length;
+
+        for (i = 0; i < poolCount; i++) {
+            var CANTO_PAIR = new ethers.Contract(pairAddresses[i], CANTO_PAIR_ABI, App.provider);
+            var CTOKEN_CONTRACT = new ethers.Contract(ctokenAddresses[i], CTOKEN_ABI_CANTO, App.provider);
+            var CANTO_ROUTER = new ethers.Contract("0xa252eee9bde830ca4793f054b506587027825a8e", CANTO_ROUTER_ABI, App.provider)
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: move outside

File: src/static/js/canto_canto.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            var token0Amount = await CANTO_PAIR.reserve0();
+            var token1Amount = await CANTO_PAIR.reserve1();
+            const decimals0 = 10 ** decimals[token0];
+            const decimals1 = 10 ** decimals[token1];
+            token0Amount = token0Amount/decimals0;
+            token1Amount = token1Amount/decimals1;
+            const TVL = token0Amount*prices[token0]["usd"] + token1Amount*prices[token1]["usd"]
+            var tokenPrice = await CANTO_ROUTER.getUnderlyingPrice(ctokenAddresses[i]);
+            // var tokenPriceMantissa = (18 - decimals[token0]) + (18 - decimals[token1]) + 18;
+            var tokenPrice = tokenPrice/(10**18); // made it 18, as for every case it will be (18 - 18) + 18
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: get this value from decimals defined above

File: src/static/js/canto_canto.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            _print(`TVL - $${formatMoney(TVL)}`)
+            // _print(`token0 amount - ${token0Amount}`)
+            // _print(`token1 amount - ${token1Amount} \n`)
+
+            const borrowRate = await CTOKEN_CONTRACT.supplyRatePerBlock();
+            const secondsInYear = 365*24*60*60;
+            const blockTime = 5.8;
+            const blocksPerYear = secondsInYear/blockTime;
+            // _print(`borrowRate- ${(borrowRate)}`)
+            const apr = borrowRate*blocksPerYear/(10**18);
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: change `apr` to `dist apr`

File: src/static/js/canto_canto.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+    const pairAddresses = [
```
Comment (zjesko) [inline src/static/js/canto_canto.js]: get pair addresses from `CTOKEN_CONTRACT.underlying()`

**Review by zjesko — CHANGES_REQUESTED** [review]

File: src/static/js/canto.js (hunk 1 above)
Comment (0xtemujin) [reply src/static/js/canto.js]: Done

File: src/static/js/canto_canto.js (hunk 2 above)
Comment (0xtemujin) [reply src/static/js/canto_canto.js]: Done

**Pull request closed without merging**